src/server/mod.rs

```diff
@@ -412,17 +410,24 @@ impl<S, B> Server<S, B>
             let addr = socket.remote_addr;
             debug!("accepted new connection ({})", addr);
 
-            let service = new_service.new_service()?;
+            let service = match new_service.new_service() {
+                Ok(service) => service,
+                Err(err) => return future::Either::Left(future::err(err))
```

Comment:
Why does this return `Either` now? It's not immediately clear to me.

Reply:
It returns either because further down we return the `Spawn` future from the same function. The `Spawn` future is lazy and needs to be polled to spawn the future (since it's using the executor from the context to spawn).

Reply:
Oh, weird! (That `spawn` API feels a little odd, but not your fault :))

Reply:
- [ ] change to use `.left()` and `.right()`